desktop_version/src/main.cpp

```diff
@@ -978,8 +982,25 @@ static enum LoopCode loop_begin(void)
         key.Poll();
     }
 
-    // Update network per frame.
-    NETWORK_update();
+    // Discord RPC handling (and maybe later Steam RPC OwO)
+    rpcArea = map.currentarea(game.roomx, game.roomy);
+    rpcRoomname = map.roomname;
+    if (game.gamestate == TITLEMODE)
+    {
+        rpcArea = "Exploring the Menus";
+        rpcRoomname = "";
+    }
+    if (game.gamestate == EDITORMODE)
+    {
+        rpcArea = "Making a Level";
+        rpcRoomname = "";
+    }
+    // Dirty fix for custom levels getting the area from Dimension VVVVVV
+    if (map.custommode) {
+        rpcArea = game.customleveltitle.c_str();
+    }
+    // Update network APIs once per frame
+    NETWORK_update(rpcArea, rpcRoomname);
```

Comment:
Rich Presence isn't a real-time-critical feature, so while I'm trapped in SDL crunch it'd be great to add a real setRichPresence API that gets called only when game state or level data changes. Best case this does a lot of unused string churning, worst case it hammers a naive network API which might get our appid rate-limited.